scripts/SANS/sans/common/file_information.py

```diff
@@ -287,30 +290,201 @@ def get_event_mode_information(file_name):
         # Open instrument group
         is_event_mode = False
         for value in list(first_entry.values()):
-            if "NX_class" in value.attrs and "NXevent_data" == value.attrs["NX_class"]:
+            if "NX_class" in value.attrs and "NXevent_data" == value.attrs["NX_class"].decode("utf-8"):
                 is_event_mode = True
                 break
     return is_event_mode
 
 
-# ---------
+# ----------------------------------------------------------------------------------------------------------------------
+# Functions for Added data
+# ----------------------------------------------------------------------------------------------------------------------
+# 1. An added file will always have a file name SOME_NAME-add.nxs, ie we can preselect an added file by the -add
+#    qualifier
+# 2. Scenario 1: Added histogram data, ie files which were added and saved as histogram data.
+# 2.1 Added histogram files will contain one or more (if they were based on multi-period files) entries in the hdfd5
+#    file where the first level entry will be named mantid_workspace_X where X=1,2,3,... . Note that the numbers
+#    correspond  to periods.
+# 3. Scenario 2: Added event data, ie files which were added and saved as event data.
+# 3.1 TODO
+
+
+def get_date_and_run_number_added_nexus(file_name):
+    with h5.File(file_name) as h5_file:
+        keys = list(h5_file.keys())
+        first_entry = h5_file[keys[0]]
+        logs = first_entry["logs"]
+        # Start time
+        start_time = logs["start_time"]
+        start_time_value = DateAndTime(start_time["value"][0])
+        # Run number
+        run_number = logs["run_number"]
+        run_number_value = int(run_number["value"][0])
+    return start_time_value, run_number_value
+
+
+def get_added_nexus_information(file_name):  # noqa
```

Comment:
I find this function difficult to read given the number of nested functions. Is there much benefit from the nesting?

Also, as general point we should keep things like `ADDED_SUFFIX = "-add_added_event_data"` at file scope as constants. I also think it would be worth defining the other string constants used in a similar manner at the top of the module to indicate they are constants.